src/adapter/graphql/src/queries/datasets/dataset_data.rs

```diff
@@ -43,6 +43,28 @@ impl<'a> DatasetData<'a> {
         Ok(dataset_statistics.data_size)
     }
 
+    /// Total number of object links in this dataset
+    #[tracing::instrument(level = "info", name = DatasetData_num_object_links, skip_all)]
+    async fn num_object_links(&self, ctx: &Context<'_>) -> Result<u64> {
```

Comment:
how about `num_linked_objects`?

As per [deduplication section of RFC](https://github.com/open-data-fabric/open-data-fabric/blob/master/rfcs/017-large-files-linking.md#accounting-for-duplicates-in-size-calculation) - we will initially provide a "naive" count, but eventually I think this API will evolve into providing accurate number of objects, accounting for duplicates and all branches - thus `num_linked_objects` and not `num_object_linkgs`.